certification/internal/policy/operator/certified_images.go

```diff
@@ -0,0 +1,104 @@
+	"github.com/operator-framework/operator-manifest-tools/pkg/pullspec"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification/pyxis"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// imageFinder interface is used for testing. It represents the FindImagesByDigest
+// function that is part of the Pyxis client.
+type imageFinder interface {
+	FindImagesByDigest(context.Context, []string) ([]pyxis.CertImage, error)
+}
+
+// CertifiedImagesCheck scans the CSV and validates that all refereenced images are certified.
+type certifiedImagesCheck struct {
+	imageFinder        imageFinder
+	nonCertifiedImages []string
+}
+
+// NewCertifiedImagesCheck returns a *certifiedImagesCheck that will scan the CSV and validate
+// that all referenced images are certified.
+func NewCertifiedImagesCheck(imageFinder imageFinder) *certifiedImagesCheck {
+	return &certifiedImagesCheck{
+		imageFinder: imageFinder,
+	}
+}
+
+func (p *certifiedImagesCheck) Validate(ctx context.Context, imgRef certification.ImageReference) (bool, error) {
+	imageDigests, err := p.dataToValidate(ctx, filepath.Join(imgRef.ImageFSPath, "manifests"))
+	if err != nil {
+		return false, err
+	}
+
+	return p.validate(ctx, imageDigests)
+}
+
+func (p *certifiedImagesCheck) dataToValidate(ctx context.Context, imagePath string) ([]string, error) {
+	operatorManifests, err := pullspec.FromDirectory(imagePath, pullspec.DefaultHeuristic)
+	if err != nil {
+		return nil, err
+	}
+	imageNames, err := image.Extract(operatorManifests)
+	if err != nil {
+		return nil, err
+	}
+
+	imageDigests := make([]string, 0, len(imageNames))
+	for _, img := range imageNames {
+		digest, err := name.NewDigest(img)
+		if err != nil {
+			log.Warningf("Image does not appear to be pinned: %s: %v", img, err)
+			p.nonCertifiedImages = append(p.nonCertifiedImages, img)
+			continue
+		}
+		imageDigests = append(imageDigests, digest.DigestStr())
+	}
+
+	return imageDigests, nil
+}
+
+func (p *certifiedImagesCheck) validate(ctx context.Context, imageDigests []string) (bool, error) {
+	pyxisImages, err := p.imageFinder.FindImagesByDigest(ctx, imageDigests)
+	if err != nil {
+		return false, err
+	}
+
+	for _, img := range pyxisImages {
+		if !img.Certified {
+			fullImg := fmt.Sprintf("%s/%s@%s", img.Repositories[0].Registry, img.Repositories[0].Repository, img.DockerImageDigest)
+			log.Warningf("Image is not certified: %s", fullImg)
+			p.nonCertifiedImages = append(p.nonCertifiedImages, fullImg)
+		}
+	}
+	return true, nil
+}
+
+func (p *certifiedImagesCheck) Name() string {
+	return "BundleImageRefsAreCertified"
+}
+
+func (p *certifiedImagesCheck) Metadata() certification.Metadata {
+	return certification.Metadata{
+		Description:      "Checking that all images referenced in the CSV are certified",
+		Level:            "optional",
+		KnowledgeBaseURL: "https://sdk.operatorframework.io/docs/olm-integration/testing-deployment/",
```

Comment:
Shouldn't these point to our policy docs?

Reply:
I would say so.